inngest/Cargo.toml

```diff
@@ -20,7 +20,10 @@ serde_json = "1.0"
 slug = "0.1.4"
 typetag = "0.2.13"
 inngest_macros = { path = "../macros" }
+futures-core = "0.3.30"
 
 # used for step encoding in SDK
 sha1 = "0.10.6"
 base16 = "0.2.1"
+futures = "0.3.30"
+parking_lot = "0.12.3"
```

Comment:
we don't need this but it's faster and easier to work with than the std lib mutex because you don't need to unwrap

Reply:
if it's not needed, it would be great to not include it.
it's best to use the least amount of packages as possible.
(I'm planning to clean up the ones above too).

Or it could be a feature dep, which can be included if user enables the feature. (e.g. `axum`)

Reply:
another thought about the mutex is, each of these execution are in their own separate threads already. so perf is not that big of a concern.
it's very unlikely that it'll be accessed by multiple threads concurrently.